python/pyarrow/array.pxi

```diff
@@ -947,10 +947,18 @@ cdef class Array(_PandasConvertible):
         if not is_primitive(self.type.id) or self.type.id == _Type_BOOL:
             raise NotImplementedError('NumPy array view is only supported '
                                       'for primitive types.')
-        buflist = self.buffers()
-        assert len(buflist) == 2
-        return np.frombuffer(buflist[-1], dtype=self.type.to_pandas_dtype())[
-            self.offset:self.offset + len(self)]
+
+        cdef:
+            PyObject* out
+            PandasOptions c_options
+            object values
+
+        c_options.zero_copy_only = True
```

Comment:
Hm. This seems too restrictive. If we can zero-copy, then let's do so, but we should not fail if it's not possible.

Reply:
I just replicated the current `to_numpy` behaviour of only allowing zero-copy conversion (the docstring says to return a numpy view on the arrow array). 

But we can certainly expand the scope, eg adding a `zero_copy_only` keyword like other conversions have?